scripts/skills/effects/rf_life_leech_effect.nut

```diff
@@ -0,0 +1,100 @@
+this.rf_life_leech_effect <- ::inherit("scripts/skills/skill", {
+	m = {
+		LifeLeechValue = 1.0,   // 1.0 equals to 100% damage leeched as life
+        LeechDistance = 1,
+		ForceHeal = false   // Temporary helper variable
+	},
+	function create()
+	{
+		this.m.ID = "effects.rf_life_leech";
+		this.m.Name = "Life Leech";
+		this.m.Description = "This character recovers Hitpoints when dealing damage to enemies with blood.";
+		this.m.Icon = "skills/status_effect_09.png";
+		this.m.IconMini = "";   // We currently have no fitting mini-icon for this
+		this.m.Overlay = "status_effect_09";
+		this.m.SoundOnUse = [
+			"sounds/enemies/vampire_life_drain_01.wav",
+			"sounds/enemies/vampire_life_drain_02.wav",
+			"sounds/enemies/vampire_life_drain_03.wav"
+		];
+		this.m.Type = ::Const.SkillType.StatusEffect;
+		this.m.Order = this.Const.SkillOrder.Perk;
+		this.m.IsActive = false;
+		this.m.IsRemovedAfterBattle = false;
+		this.m.IsStacking = false;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getTooltip();
+
+        local withinText = "adjacent enemies";
+        if (this.m.LeechDistance > 1) withinText = "enemies within " + ::MSU.Text.colorGreen(this.m.LeechDistance) + " tiles";
+        ret.push({
+            id = 11,
+            type = "text",
+            icon = "ui/icons/health.png",
+            text = "Heal " + ::MSU.Text.colorGreen((this.m.LifeLeechValue * 100) + "%") + " of Hitpoint damage inflicted on " + withinText + " that have blood"
+        });
+
+		return ret;
+	}
+
+	function onCombatStarted()
+	{
+		this.m.ForceHeal = false;
+	}
+
+	function onCombatFinished()
+	{
+		this.m.ForceHeal = false;
+	}
+
+	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	{
+		if (_damageInflictedHitpoints <= 0) return;
```

Comment:
If the damage is negative, it seems like a bug. Therefore, it is preferable to use `==` here to show the intent clearly.

Reply:
I agree that negative damage dealt should be considered a bug.

I was copying vanilla in this case. In the end I didn't write the vanilla code and don't know for sure whether negative values are possible.

Reply:
It can never be negative, in vanilla. actor.nut `onDamageReceived` function:
```squirrel
damage = this.Math.max(0, this.Math.max(this.Math.round(damage), this.Math.min(this.Math.round(_hitInfo.DamageMinimum), this.Math.round(_hitInfo.DamageMinimum * p.DamageReceivedTotalMult))));
_hitInfo.DamageInflictedHitpoints = damage;
```
which is then passed to the `onTargetHit` skill event in `skill.nut` function `onScheduledTargetHit`:
```squirrel
_info.Container.onTargetHit(_info.Skill, _info.TargetEntity, hitInfo.BodyPart, hitInfo.DamageInflictedHitpoints, hitInfo.DamageInflictedArmor);
```